packages/v4/src/content/design-guidelines/components/tree-view/tree-view.md

```diff
@@ -3,5 +3,91 @@ id: Tree view
+3. **Child node:** A node within the parent node (for example, a document within a folder)
+4. **Leaf node:** A node without children
+5. **Truncation indicator:** Ellipses indicating truncated text for longer names that don't fit in the node width
+
+## Usage
+
+A tree view can be used for: 
+1. Changing the content on a page based on the selected item. 
+2. Selecting items or applying filters. 
+
+<img src="./img/Default-tree-view.png" width="1004" />
+
+Use a tree view when:   
+* Selecting is the main use-case. 
+* You apply filters that need to be structured in a clear hierarchy. 
+* The data is structured into levels.
+* You need to display hierarchies that have more than 2 levels.
+
+Don’t use a tree view when: 
+* You select 1 item from a small set. Instead, use [simple list](/components/simple-list), [select lists](/components/select), [radio buttons](/components/radio), [checkboxes](/components/checkbox), or a [dropdown](/components/dropdown).
+* You need to see an overview of a large amount of data.
+* You need to compare two large sets of data. Use [charts](/charts/about) for a better view. 
+* The hierarchy is too deep and includes hundreds of items. Instead, use [drilldown menu](/components/menu#drilldown).
+
+## Tree view capabilities
+
+You can add these functionalities to a tree view:
+* **[Search bar](#adding-a-search-bar-to-a-tree-view)** for searching items in the tree view
+* **[Checkboxes](#adding-checkboxes-to-a-tree-view)** for selecting items in the tree view
+* **[Badges](#adding-badges-to-a-tree-view)** for showing the number of child nodes in the tree view
+* **[Icons](#adding-icons-to-a-tree-view)** for visually representing the node types in the tree view
+* **[Action items](#adding-action-items-to-a-tree-view)** for adding an action to each item in the tree view
+
+### Adding a search bar to a tree view
+
+Add a search bar to a tree view if you're including a large amount of data and need to provide users with a quick way to find a single item. If your dataset is small and easy to navigate, then you don't need to add a search bar.
+
+<img src="./img/with-search-bar.png" width="940" />
+
+### Adding checkboxes to a tree view
+
+Add checkboxes to a tree view if you’re selecting multiple items or applying multiple filters. Selecting should be a primary function of your use-case. If your dataset is small and you select just 1 item, use [select lists](/components/select), [radio buttons](/components/radio), [checkboxes](/components/checkbox), or a [dropdown](/components/dropdown) instead. 
+
+<img src="./img/with-checkboxes.png" width="932" />
+
+### Adding badges to a tree view
+
+Add badges to a tree view if you want to show the number of items inside. Don’t use it for displaying different information. Items without children can't have badges as they don't contain anything.
+
+<img src="./img/with-badges.png" width="332" />
+
+### Adding icons to a tree view
```

Comment:
Just a small comment - to be it seems weird to add a folder icon in a child node, because it's not a folder haha. Is it **required*** that if you use icons, you HAVE to use them on everything?

Reply:
I agree, but it can be an icon placeholder. I changed the icons to be more relevant now. 
And I think that it should be defined: 
- use it everywhere or you can leave the leaf nodes/items without icons. But, don't combine these leaf items with icons and without. 
1st option: 
![with-icons](https://user-images.githubusercontent.com/32618096/97865647-d546d680-1d0a-11eb-92b6-b771d0800424.png)

2nd option:
![Default with icons tree view](https://user-images.githubusercontent.com/32618096/97865690-e1329880-1d0a-11eb-85ce-c9abda87cbcd.png)

Reply:
I've changed the image and the sentence: _If you choose to add icons to a tree view, apply the icons on all items in a tree or leave the items without children empty, to maintain consistency._

Don't know if the word "empty" is right.

Reply:
You could use "iconless" instead of empty! Do you know from a technical stand point if it's possible to have icons on some nodes and not others within the same tree?

Reply:
I'll check it with Michael Coker to be 100% sure.

Reply:
Checked with Michael > it should be fine :)

Reply:
Great!